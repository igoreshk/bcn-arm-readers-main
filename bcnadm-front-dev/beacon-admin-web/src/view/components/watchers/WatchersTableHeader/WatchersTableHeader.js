import React from 'react';
import { TableRow, TableCell, Button } from '@material-ui/core';
import Filter from '@material-ui/icons/FilterList';
import SortedAsc from '@material-ui/icons/KeyboardArrowDown';
import SortedDesc from '@material-ui/icons/KeyboardArrowUp';
import PropTypes from 'prop-types';
import { SORTED as watchersSort } from 'src/utils/watchersSort';

const Unsorted = () => <SortedAsc style={{ fill: '#CCCCCC' }} />;

const headerStyle = {
  color: 'rgba(0, 0, 0, 0.54)',
  cursor: 'pointer',
  display: 'flex',
  alignItems: 'center'
};

const getIconByField = (sortedBy, field) => {
  if (sortedBy === watchersSort[field].desc) {
    return <SortedDesc />;
  }
  if (sortedBy === watchersSort[field].asc) {
    return <SortedAsc />;
  }
  return <Unsorted />;
};

export const WatchersTableHeader = (props) => {
  return (
    <TableRow>
      <TableCell>
        <div
          onClick={() => {
            props.toggleSortWatchersByField('name');
          }}
          style={headerStyle}
          className="name"
        >
          {props.translate('watchers.watcherName')}
          {getIconByField(props.sortedBy, 'name')}
        </div>
      </TableCell>
      <TableCell>
        <div
          onClick={() => {
            props.toggleSortWatchersByField('visitors');
          }}
          style={headerStyle}
          className="visitors"
        >
          {props.translate('watchers.visitor')}
          {getIconByField(props.sortedBy, 'visitors')}
        </div>
      </TableCell>
      <TableCell className="remove">
        <Button onClick={props.toggleFilter} className="removeButton">
          <Filter />
          <span style={{ marginLeft: '10px' }}>{props.translate('visitors.filter')}</span>
        </Button>
      </TableCell>
    </TableRow>
  );
};

WatchersTableHeader.propTypes = {
  sortedBy: PropTypes.string,
  toggleFilter: PropTypes.func.isRequired,
  toggleSortWatchersByField: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired
};

export default WatchersTableHeader;
