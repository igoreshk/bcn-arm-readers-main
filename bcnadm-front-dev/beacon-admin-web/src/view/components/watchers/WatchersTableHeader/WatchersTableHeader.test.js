import React from 'react';
import { shallow } from 'enzyme';
import { WatchersTableHeader } from './WatchersTableHeader';

const props = {
  toggleFilter: jest.fn(),
  toggleSortWatchersByField: jest.fn(),
  translate: jest.fn()
};

const wrapper = shallow(<WatchersTableHeader {...props} />);

describe('WatchersTableHeader', () => {
  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());

  it('Check toggleSortWatchersByField was called with "visitors"', () => {
    wrapper.find('.visitors').simulate('click');
    expect(props.toggleSortWatchersByField).toHaveBeenCalledWith('visitors');
  });

  it('Check toggleSortWatchersByField was called with "name"', () => {
    wrapper.find('.name').simulate('click');
    expect(props.toggleSortWatchersByField).toHaveBeenCalledWith('name');
  });
});
