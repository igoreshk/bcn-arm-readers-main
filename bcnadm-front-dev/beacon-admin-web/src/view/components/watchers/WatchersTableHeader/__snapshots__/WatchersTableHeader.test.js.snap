// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchersTableHeader Check matching snapshot 1`] = `
<WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableCell))>
    <div
      className="name"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "color": "rgba(0, 0, 0, 0.54)",
          "cursor": "pointer",
          "display": "flex",
        }
      }
    >
      <Unsorted />
    </div>
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    <div
      className="visitors"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "color": "rgba(0, 0, 0, 0.54)",
          "cursor": "pointer",
          "display": "flex",
        }
      }
    >
      <Unsorted />
    </div>
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))
    className="remove"
  >
    <WithStyles(ForwardRef(Button))
      className="removeButton"
      onClick={[MockFunction]}
    >
      <Memo(ForwardRef(FilterListIcon)) />
      <span
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(TableCell))>
</WithStyles(ForwardRef(TableRow))>
`;
